packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -67,7 +67,7 @@ ${formatPluginOptionsForCLI(pluginConfig.getOriginalPluginOptions(), errors)}`)
 
   let currentSyncData
   try {
-    let query = syncToken ? { nextSyncToken: syncToken } : { initial: true }
+    let query = { initial: true } // always initial to be able to resolve links
```

Comment:
Hey @Khaledgarbaya, what is the impact of this change on spaces with _a lot_ of content?

Reply:
I am trying to find another solution for this but for now, this will pull all the data.

Reply:
@Khaledgarbaya This does seem problematic for very large spaces. Is this fix worth this impact or shall we wait for a better solution? 

@pieh Thoughts?

Reply:
@sidharthachatterjee I think so too. I will try probably this week another approach and see.